### ISSUE TITLE
Setting `karpenter.sh/provisioner-name` to nodeAffinity causes the pod to be PENDING

### ISSUE THREAD

**Issue by kyontan** [aws/karpenter#2320]
## Version

**Karpenter**: v0.14.0 and v0.15.0 (later than v0.13.2)

**Kubernetes**: v1.22.10 (`version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.10-eks-84b4fe6", GitCommit:"cc6a1b4915a99f49f5510ef0667f94b9ca832a8a", GitTreeState:"clean", BuildDate:"2022-06-09T18:24:04Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}`)

## Expected Behavior

As written in the docs (https://karpenter.sh/v0.15.0/faq/#if-multiple-provisioners-are-defined-which-will-my-pod-use), we use a feature to specify specific provisioner for the pods.

By writing manifest like below:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
  namespace: default
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: karpenter.sh/provisioner-name
                operator: In
                values:
                - default-provisioner
      containers:
      - image: nginx
        imagePullPolicy: IfNotPresent
        name: nginx
        resources:
          requests:
            cpu: 4
      tolerations:
      - effect: NoSchedule
        key: provisioned-by-karpenter
        operator: Exists
```

(If the resource request is satisfied to already existing nodes, Karpenter skips scheduling, so that the resource requests should be large enough to trigger Karpenter to create new node.)

## Actual Behavior

The pods are worked as expected before v0.14.0, but the pods got `PENDING` status since v0.14.0 (and later).

The `karpenter` pod shows the DEBUG log:

```
controller 2022-08-19T04:45:17.747Z    DEBUG    controller.provisioning    Ignoring pod, label karpenter.sh/provisioner-name is restricted; specify a well known label: [karpenter.k8s.aws/instance-category karpenter.k8s.aws/instance-cpu karpenter.k8s.aws/instance-family karpenter.k8s.aws/instance-generation karpenter.k8s.aws/instance-gpu-count karpenter.k8s.aws/instance-gpu-manufacturer karpenter.k8s.aws/instance-gpu-memory karpenter.k8s.aws/instance-gpu-name karpenter.k8s.aws/instance-hypervisor karpenter.k8s.aws/instance-local-nvme karpenter.k8s.aws/instance-memory karpenter.k8s.aws/instance-pods karpenter.k8s.aws/instance-size karpenter.sh/capacity-type kubernetes.io/arch kubernetes.io/os node.kubernetes.io/instance-type topology.kubernetes.io/region topology.kubernetes.io/zone], ora custom label that does use a restricted domain: [k8s.io karpenter.k8s.aws karpenter.sh kubernetes.io]    {"commit": "3d87474", "pod": "XXX"}
```

## Steps to Reproduce the Problem

- Create a pod (or rollout, deployment, ...) with nodeAffinity that uses a key `karpenter.sh/provisioner-name`.

## Resource Specs and Logs

I think just setting nodeAffinity is the cause, but if needed, please let me know!

**Comment by ellistarn** [comment]
This is a big issue! Thanks for the report.

**Comment by ellistarn** [comment]
Fixed in v0.16.
